File: ext/lzws/src/lzws_ext/file.c
// Ruby bindings for lzws library.
// Copyright (c) 2019 AUTHORS, MIT License.

#include <lzws/file.h>

#include "lzws_ext/common.h"
#include "lzws_ext/file.h"
#include "lzws_ext/macro.h"
#include "ruby.h"

VALUE lzws_ext_compress_file(VALUE LZWS_EXT_UNUSED(self), VALUE source, VALUE destination, VALUE options)
{
  Check_Type(source, T_FILE);
  Check_Type(destination, T_FILE);

  LZWS_EXT_GET_COMPRESSOR_OPTIONS(options);

  return Qnil;
}

VALUE lzws_ext_decompress_file(VALUE LZWS_EXT_UNUSED(self), VALUE source, VALUE destination, VALUE options)
{
  Check_Type(source, T_FILE);
  Check_Type(destination, T_FILE);

  LZWS_EXT_GET_DECOMPRESSOR_OPTIONS(options);

  return Qnil;
}

File: lib/lzws/file.rb
# Ruby bindings for lzws library.
# Copyright (c) 2019 AUTHORS, MIT License.

require "adsp/file"
require "lzws_ext"

require_relative "option"

module LZWS
  # LZWS::File class.
  class File < ADSP::File
    # Current option class.
    Option = LZWS::Option

    def self.native_compress_io(*args)
      LZWS._native_compress_io(*args)
    end

    def self.native_decompress_io(*args)
      LZWS._native_decompress_io(*args)
    end
  end
end
